From 4e4ca032a8a3e03cab378666d0edacdfd89aeec4 Mon Sep 17 00:00:00 2001
From: Shivani Raval <sraval@habana.ai>
Date: Thu, 22 May 2025 18:25:11 +0300
Subject: [PATCH] [JSW-0] jgs: macro modifications for jgs

Removed RST Macros as we are using RESET macros instead of RST.
Added the required macros for SW mailbox support for JGS.

for more info: https://gerrit.habana-labs.com/#/c/464831/

Change-Id: Ia04f2faa8b136b0fab4ff1a0cbd97591cebb068a
Signed-off-by: Shivani Raval <sraval@habana.ai>
---

diff --git a/fs/jgs/jgs_mem_addr_dt.h b/fs/jgs/jgs_mem_addr_dt.h
index 2d187fd..8f01f40 100644
--- a/fs/jgs/jgs_mem_addr_dt.h
+++ b/fs/jgs/jgs_mem_addr_dt.h
@@ -22,7 +22,7 @@
 /*
  * ToDo: Remove these macros after successful merge of
  *       io die and compute die macros for preboot.
-*/
+ */
 
 #define ARC_PREBOOT_ICCM_OFFSET	0
 #define ARC_PREBOOT_ICCM_SIZE	0x100
@@ -48,12 +48,6 @@
 #define ARC_DSM_ICCM_SIZE	0x100
 #define ARC_DSM_DCCM_OFFSET	80000000
 #define ARC_DSM_DCCM_SIZE	0x100
-
-/* TODO: remove _RST_ macros */
-#define ARC_RST_ICCM_OFFSET	0
-#define ARC_RST_ICCM_SIZE	0x40
-#define ARC_RST_DCCM_OFFSET	80000000
-#define ARC_RST_DCCM_SIZE	0x20
 
 #define ARC_RESET_ICCM_SECTION	1
 #define ARC_RESET_DCCM_SECTION	1
@@ -149,13 +143,13 @@
 #define SPI_FLASH_REGION_RESERVED_2_SIZE			b0000
 
 /*
- * FS1 Dup engine supports two regions with 8 groups each. Region 0
- * On IO die, Region 0 is allocated to mgmt arc, Region 1 is
- * allocated to pid Arc.
- * On Compute die, Region 0 is allocated to mgmt arc, Region 1 is
- * not used
- * DUP_GROUP_MMIO_RANGE = 512, DUP_NUM_GROUPS_PER_REGION =8
- */
+* FS1 Dup engine supports two regions with 8 groups each. Region 0
+* On IO die, Region 0 is allocated to mgmt arc, Region 1 is
+* allocated to pid Arc.
+* On Compute die, Region 0 is allocated to mgmt arc, Region 1 is
+* not used
+* DUP_GROUP_MMIO_RANGE = 512, DUP_NUM_GROUPS_PER_REGION =8
+*/
 #define DUP_ENGINE_IO_REGION_0_OFFSET		f9cfc000
 #define DUP_ENGINE_IO_REGION_1_OFFSET		f9cfd000
 
@@ -240,4 +234,44 @@
 /* CMPT1 MGMT --> CMPT0 RESET */
 #define COMPUTE1_SW_MAILBOX1_1					F3CA8004
 
+/* IO: MGMT <==> PID IPM */
+#define SPSRAM_IPM_BUFF_SIZE			0x100
+#define SPSRAM_IO_MGMT_TX_IO_PID_RX_MBX		0x78A00000
+#define SPSRAM_IO_PID_TX_IO_MGMT_RX_MBX		(SPSRAM_IO_MGMT_TX_IO_PID_RX_MBX + \
+						SPSRAM_IPM_BUFF_SIZE) /* 0x78A00100 */
+
+/* IO: MGMT <==> Reset IPM */
+#define SPSRAM_IO_MGMT_TX_IO_RST_RX_MBX		(SPSRAM_IO_PID_TX_IO_MGMT_RX_MBX + \
+						SPSRAM_IPM_BUFF_SIZE) /* 0x78A00200 */
+#define SPSRAM_IO_RST_TX_IO_MGMT_RX_MBX		(SPSRAM_IO_MGMT_TX_IO_RST_RX_MBX + \
+						SPSRAM_IPM_BUFF_SIZE) /* 0x78A00300 */
+
+/* IO: MGMT <==> MISC IPM */
+#define SPSRAM_IO_MGMT_TX_IO_MISC_RX_MBX	(SPSRAM_IO_RST_TX_IO_MGMT_RX_MBX + \
+						SPSRAM_IPM_BUFF_SIZE) /* 0x78A00400 */
+#define SPSRAM_IO_MISC_TX_IO_MGMT_RX_MBX	(SPSRAM_IO_MGMT_TX_IO_MISC_RX_MBX + \
+						SPSRAM_IPM_BUFF_SIZE) /* 0x78A00500 */
+
+/* IO MGMT <==> COMPUTE0 MGMT IPM */
+#define SPSRAM_COMPUTE0_BASE			0x70A00000
+#define SPSRAM_IO_MGMT_TX_CMPT0_MGMT_RX_MBX	SPSRAM_COMPUTE0_BASE
+#define SPSRAM_CMPT0_MGMT_TX_IO_MGMT_RX_MBX	(SPSRAM_IO_MGMT_TX_CMPT0_MGMT_RX_MBX + \
+						SPSRAM_IPM_BUFF_SIZE) /* 0x70A00100 */
+/* COMPUTE0: MGMT <==> Reset IPM */
+#define SPSRAM_CMPT0_MGMT_TX_CMPT0_RST_RX_MBX	(SPSRAM_CMPT0_MGMT_TX_IO_MGMT_RX_MBX + \
+							SPSRAM_IPM_BUFF_SIZE) /* 0x70A00200 */
+#define SPSRAM_CMPT0_RST_TX_CMPT0_MGMT_RX_MBX	(SPSRAM_CMPT0_MGMT_TX_CMPT0_RST_RX_MBX + \
+						SPSRAM_IPM_BUFF_SIZE) /* 0x70A00300 */
+
+/* IO MGMT <==> COMPUTE1 MGMT IPM */
+#define SPSRAM_COMPUTE1_BASE			0x72A00000
+#define SPSRAM_IO_MGMT_TX_CMPT1_MGMT_RX_MBX	SPSRAM_COMPUTE1_BASE
+#define SPSRAM_CMPT1_MGMT_TX_IO_MGMT_RX_MBX	(SPSRAM_IO_MGMT_TX_CMPT1_MGMT_RX_MBX + \
+						SPSRAM_IPM_BUFF_SIZE) /* 0x72A00100 */
+/* COMPUTE1: MGMT <==> Reset IPM */
+#define SPSRAM_CMPT1_MGMT_TX_CMPT1_RST_RX_MBX	(SPSRAM_CMPT1_MGMT_TX_IO_MGMT_RX_MBX + \
+							SPSRAM_IPM_BUFF_SIZE) /* 0x72A00200 */
+#define SPSRAM_CMPT1_RST_TX_CMPT1_MGMT_RX_MBX	(SPSRAM_CMPT1_MGMT_TX_CMPT1_RST_RX_MBX + \
+						SPSRAM_IPM_BUFF_SIZE) /* 0x72A00300 */
+
 #endif
